lib/smart_proxy.rb

```diff
@@ -136,24 +128,53 @@ def load_ssl_certificate(path)
       raise e
     end
 
+    def pid_status(pid_path)
+      return :exited unless File.exist?(pid_path)
+      pid = ::File.read(pid_path).to_i
+      return :dead if pid == 0
+      Process.kill(0, pid)
+      :running
+    rescue Errno::ESRCH
+      :dead
+    rescue Errno::EPERM
+      :not_owned
+    end
+
+    def check_pid
+      case pid_status(pid_path)
+      when :running, :not_owned
+        logger.error "A server is already running. Check #{pid_path}"
+        exit(1)
+      when :dead
+        File.delete(pid_path)
+      end
+    end
+
+    def write_pid
+      FileUtils.mkdir_p(File.dirname(pid_path)) unless File.exist?(pid_path)
+      File.open(pid_path, ::File::CREAT | ::File::EXCL | ::File::WRONLY){|f| f.write("#{Process.pid}") }
+      at_exit { File.delete(pid_path) if File.exist?(pid_path) }
+    rescue Errno::EEXIST
```

Comment:
Ahh, so this is to handle a concurrent launch of multiple instances of smart-proxy? Could you add a comment here explaining this pls?